web/src/engine/websites/RuManhua.ts

```diff
@@ -0,0 +1,92 @@
+import { Tags } from '../Tags';
+import icon from './RuManhua.webp';
+import { Chapter, DecoratableMangaScraper, Manga, type MangaPlugin } from '../providers/MangaPlugin';
+import * as Common from './decorators/Common';
+import { FetchCSS, FetchJSON } from '../platform/FetchProvider';
+
+type APIResponse<T> = {
+    data : T | string
+}
+
+type APIManga = {
+    id: string,
+    bookName: string
+}
+
+type APIChapter = {
+    chapterid: string,
+    chaptername: string
+}
+
+const pagesScript = `
+    new Promise (resolve => resolve( [... document.querySelectorAll('div.chapter-img-box img')].map(image=> image.dataset.src || image.src)));
```

Comment:
This should work without wrapping the result in a promise, since it is a single command